### PR TITLE
example of querying pair reserves, add mainnet to hardhat config

### PR DESCRIPTION
**Pull request by wildmolasses**
sunday!

### CODE REVIEW

**Comment by Scapestudio** [comment]
Great talk the other day, I've learned a lot

**Review by Scapestudio — COMMENTED** [review]
Can we use the same method to query in to the sushiswap exchange?

**Comment by wildmolasses** [comment]
hey @Scapestudio , it seems you closed this branch. Can you reopen it? If you close it, the changes in this branch will never make it to "main"!

> Can we use the same method to query in to the sushiswap exchange?

Yes

**Comment by Scapestudio** [comment]
I'm sorry I'm making a mess. 
Ok so I can try todo the same thing for the sushiswap exchange if that's fine with you guys

**Comment by wildmolasses** [comment]
sure @Scapestudio 

The way we normally build on code and divide work is with issues and pull requests.

First we should make an issue that describes exactly what we need done. In this case it can be "add sushiswap example below uniswap one."

At the same time we should merge the branch in this PR (**pair-stuff-1**) so that its commits are on **main**.

Then you can make a branch (maybe call it **sushiswap-example**) using `git checkout -b sushiswap-example`. But the context in which you run this command is very important. Let's see what happens when you run `git checkout -b sushiswap example` in a few different scenarios:

- scenario 1: we didn't merge **pair-stuff-1** onto **main** yet
  - this means your local main branch will not have the commits from **pair-stuff-1** (which is the `provider.on` stuff). But eventually we will merge it. When **pair-stuff-1** gets merged, now your **sushiswap-example** branch is missing commits. You will have to perform a rebase. It is more advanced stuff, so not recommended.
- scenario 2: we did merge **pair-stuff-1** onto **main**, but you did not pull the changes down from the remote to local
  - this means your local "learning-uniswap" repo is not in sync with what's on github. So again you will have missing commits, and a rebase will be required. This is also a common situation, but if we can avoid it, we should.
- scenario 3: we merged **pair-stuff-1** onto **main**, and then you ran `git pull` to get the latest commits locally
  - good, this is the best scenario. This is a good place to branch from. You can branch from **main** knowing that main has the latest commits.

**Comment by wildmolasses** [comment]
@Scapestudio @kupcha if you guys could take a look at the changes in this PR (check the Files Changed tab) and make sure that nothing slipped in that didn't belong, then that would be good. You can "approve" this PR by going to that tab and clicking "Start a Review" and then "Approve." You can also comment on certain lines of code if you want. Maybe this PR is a good one to try it. Why don't you each try leaving a comment on some lines of code in your review, and then at the end, "approve" the PR in your review.

File: tsconfig.json [hunk 1]
```diff
@@ -18,5 +18,5 @@
   },
   "exclude": ["node_modules"],
   "files": ["./hardhat.config.ts"],
-  "include": ["tasks/**/*", "test/**/*", "types/**/*"]
+  "include": ["tasks/**/*", "test/**/*", "types/**/*", "src/**/*"]
```
Comment (Scapestudio) [inline tsconfig.json:21]: My only concern is this line of code that I don't quite understand

**Review by Scapestudio — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (wildmolasses) [reply tsconfig.json:21]: typescript has to know where our .ts files are. This array that we provide to the "include" property specifies where the ts files in our project are. Before, typescript didnt know we had a src folder. Now we're telling it, "check everything inside src and do typescript magic on it"

Comment (Scapestudio) [reply tsconfig.json:21]: Ok got it

**Review by kupcha — APPROVED** [review]
I don't see any milk in this code which is troubling but everything else is good

**Pull request merged**